src/librustc/mir/interpret/mod.rs

```diff
@@ -156,31 +156,31 @@ impl<'tcx> Pointer {
         Pointer { alloc_id, offset }
     }
 
-    pub fn wrapping_signed_offset<C: HasDataLayout>(self, i: i64, cx: C) -> Self {
+    pub fn wrapping_signed_offset(self, i: i64, cx: impl HasDataLayout) -> Self {
```

Comment:
cc @nikomatsakis I think this looks amazing and we should just change everything throughout!